src/matcher.rs

```diff
@@ -2,6 +2,7 @@
 // this module is called matcher
 
 #![allow(clippy::module_name_repetitions)]
```

Comment:
I want to maintain no clippy warnings. To remove this one, I either need to rename the module from `matcher` to something else, or change the `struct`s to not have `Matcher` as a prefix or suffix. Which would you prefer?

Note that we had no issue when the module was named `ast_matcher`.

Reply:
I'll use the macros in the parser tests and we can drop the matcher suffix. All this will do is move the line noise though.

Reply:
Unconvinced by the value in this clippy warning.